core/wallet/transaction.go

```diff
@@ -50,10 +56,16 @@ func Deposit(ledgerAddr []byte, amount int64, privateKey *ecdsa.PrivateKey, runD
 	}
 	log.Debug(fmt.Sprintf("Call Deposit API success, id: [%d]", prepareResponse.GetId()))
 
+	err = PersistTx(n.Repo.Datastore(), n.Identity.Pretty(), prepareResponse.GetId(),
+		amount, BttWallet, InAppWallet, StatusPending)
+	if err != nil {
+		log.Error(fmt.Sprintf("Record deposit tx failed, reasons: [%v]", err))
```

Comment:
return err? (this is the same as "lost" on UI ... better warn user, log.Error does not return err back to UI)

Reply:
my concern is if we return err for PersistTx, client side may do deposit/withdraw again, but actually it's already done.